python/rascaline-torch/tests/autograd.py

```diff
@@ -0,0 +1,241 @@
+import warnings
+
+import ase
+import pytest
+import torch
+
+import rascaline.torch
+from rascaline.torch import SoapPowerSpectrum, SphericalExpansion, System
```

Comment:
You mean a basic system with minimal functionality? We already have [SystemBase](https://luthaf.fr/rascaline/latest/references/api/python/systems.html#rascaline.SystemBase) in rascaline for custom system, which can be used like rascaline-torch system is used.

Reply:
Yes, but SystemBase is very rigid. The species, positions and cell is fixed. Here you can give this as arguments, which would make the import of `ase` in some more cases superflous.

Reply:
> The species, positions and cell is fixed.

No? SystemBase is a base class, if you want to reproduce the functionality of the `rascaline.torch.System` you can do something like 

```py
class System(SystemBase):
    def __init__(self, species, positions, cell):
        self.species = species
        self.positions = positions
        self.cell = cell
        # TODO: size validation for all data

    def size(self):
        return len(self.species)

    def species(self):
        return self.species

    def positions(self):
        return self.positions

    def cell(self):
        return self.cell

    def compute_neighbors(self, cutoff):
        raise NotImplementedError("this can not be used with use_native_systems=True")

    def pairs(self):
        raise NotImplementedError("this can not be used with use_native_systems=True")

    def pairs_containing(self, center):
        raise NotImplementedError("this can not be used with use_native_systems=True")
```

We could have this code as an example in SystemBase, but it feels to me it will not be that useful, since most people will use the ASE or chemfiles systems implementation.

We can and should use things like this for tests, we already do it in some places: https://github.com/Luthaf/rascaline/blob/9a049175be29a679ad954ce83b577c3a5b2cdc1e/python/rascaline/tests/test_systems.py#L4

Reply:
Added documentation in #217